src/engine/enginebuffer.cpp

```diff
@@ -587,6 +586,7 @@ void EngineBuffer::slotTrackLoaded(TrackPointer pTrack,
         m_iSeekPhaseQueued = 0;
     }
 
+    m_pCurrentTrack = pTrack;
```

Comment:
This change in EngineBuffer is a bit brittle. Now `EngineBuffer::isTrackLoaded()` will return false after `notifyTrackLoaded(pTrack, pOldTrack);` and the "track_loaded" CO is set to 1. 

Can you explain the race condition a bit more? Maybe we find a solution closer to the failing test.

Reply:
`EngineBuffer::isTrackLoaded()` will return `true` but the color won't yet have been parsed till `notifyTrackLoaded` has emitted the signal to get the `BasePlayer` to proceed the stem color.
Since `notifyTrackLoaded` doesn't use `m_pCurrentTrack` directly, I thought it was fine to change.

So basically the tests probe for `engine` load instead of `player` load.